modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/client/OpcUaSubscriptionConsumer.java

```diff
@@ -132,11 +129,22 @@ public void accept(final UaSubscription uaSubscription) {
                                     "')");
                         }
                     }
-                    resultFuture.complete(null);
-                })
-                .exceptionally(monitorThrowable -> {
-                    resultFuture.completeExceptionally(monitorThrowable);
-                    return null;
+                    return new SubscriptionResult(subscription, uaSubscription, this);
                 });
     }
+
+    public static class SubscriptionResult {
+        public final @NotNull OpcUaToMqttMapping subscription;
+        public final @NotNull UaSubscription uaSubscription;
+        public final @NotNull OpcUaSubscriptionConsumer consumer;
+
+        public SubscriptionResult(
+                @NotNull final OpcUaToMqttMapping subscription,
```

Comment:
final @NotNull for consistency would be nice

Reply:
fixed